test-requirements.txt

```diff
@@ -94,8 +75,9 @@ parso==0.8.3
     # via jedi
 pathspec==0.12.1
     # via black
+pip==24.0
+    # via pip-tools
 pip-tools==7.3.0
-    # via -r test-requirements.in
```

Comment:
Let's remove `pip-tools` from `test-requirements.in` and replace that with `uv`. Then we can remove the `pip install uv` at the top of ci.sh

Reply:
Done in [a20b4ce](https://github.com/python-trio/trio/pull/2958/commits/a20b4ce38866f24e41b5fd181a165cfafd14a4c2)